sdk/textanalytics/Azure.AI.TextAnalytics.Legacy.Shared/src/Generated/TextAnalyticsRestClient.cs

```diff
@@ -75,7 +75,7 @@ public async Task<ResponseWithHeaders<TextAnalyticsAnalyzeHeaders>> AnalyzeAsync
                 case 202:
                     return ResponseWithHeaders.FromValue(headers, message.Response);
                 default:
-                    throw await ClientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
```

Comment:
I wonder if this change means that we should revisit passing around `ClientDiagnostics` through the TA layers.   IIRC, we do this only because of this use. 
@maririos - thoughts?

Reply:
we would still need it for the LRO classes as we manage throwing exceptions there, no?

Reply:
The updated pattern doesn't require ClientDiagnostics. I just updated `LanguageServiceClient` with the new approach [here](https://github.com/Azure/azure-sdk-for-net/pull/29625/commits/deef24801d5fd27fe6da328158673599ac8cdfa6)